### DISCUSSION TITLE
No Bug: Best practices to upgrade PnP Search WP higher version in Prod environment

### DISCUSSION THREAD

**Discussion by shariq1379** [microsoft-search/pnp-modern-search#2292] [Q&A]
### Check existing issues before logging new ones
Sorry, I don't know where I should post questions to ask my question, so I just created this ticket.

Actually we are using PnP version 4.1.0.0 which we want to upgrade latest version 4.7.0.0. These are a few question which we want to know before upgrade in order to prevent risk to upgrade higher version. Your answers will be appreciated. Thanks!!! :)  

**Questions**
What best practice we should follow to upgrade latest version? is there any documentation which we can follow?
Is there any risk to upgrade? New version could introduce new issues or break existing functionality? 
How frequently we should upgrade latest version? such as quarterly/yearly. Any suggestion

**Version used**
Ex: 4.1.0.0

**Describe the bug**
No bug. Sorry to post questions here.

**Expected behavior**
NA

**Screenshots**
NA

**Desktop (please complete the following information):**
 NA

**Additional context**
NA

**Comment by wobba** [comment]
Normally you can just upgrade from one version to another by uploading a new sppkg.

There should not be any breaking changes in v4. The best is still to have test tenant or site collection with the same version you have in prod, and then update there and verify before going live with a new version.
